elyra/pipeline/processor_kfp.py

```diff
@@ -68,10 +69,14 @@ def process(self, pipeline):
 
             self.log.debug("Creating temp directory %s", temp_dir)
 
+            engine = runtime_configuration.metadata.get('engine')
             # Compile the new pipeline
             try:
                 pipeline_function = lambda: self._cc_pipeline(pipeline, pipeline_name)  # nopep8 E731
-                kfp.compiler.Compiler().compile(pipeline_function, pipeline_path)
+                if ('Argo' == engine):
+                    kfp.compiler.Compiler().compile(pipeline_function, pipeline_path)
+                else:
+                    kfp_tekton.compiler.TektonCompiler().compile(pipeline_function, pipeline_path)
             except Exception as ex:
                 raise RuntimeError('Error compiling pipeline {} at {}'.
```

Comment:
Since the `else` statement is non-specific, I would suggest updating the error message to include the `engine` value, just in case some bogus value (like 'argo' :smile: ) sneaks in.
```suggestion
                raise RuntimeError('Error compiling pipeline {} for engine '{}' at {}'.
```
Might as well convert this to an f-string if you decide to update it.